src/main/models/terminal/Terminal.java

```diff
@@ -0,0 +1,46 @@
+package main.models.terminal;
+
+import main.models.terminal.interfaces.Display;
+import main.models.vehicles.Train;
+import main.models.vehicles.TrainSize;
+
+public class Terminal implements Display {
+    private final Train train;
+    private final String terminalName;
+
+    private Terminal(final TerminalBuilder builder){
+       this.train = new Train(builder.trainSize);
+       this.terminalName = builder.terminalName;
+    }
+
+    @Override
+    public void displayProfit() {
+        System.err.println(this.train +  " has earned " + this.train.getProfit() + " kn.\n");
+    }
+
+    public Train getTrain() {
+        return train;
+    }
+
+    public static class TerminalBuilder {
```

Comment:
Just a note: in real world, you would most likely use `@Builder` from [lombok](https://projectlombok.org/features/Builder) ;)

Reply:
Thanks, but i wanted to train the builder pattern in vanilla style